examples/tutorials/nb_python/Habitat_Interactive_Tasks.py

```diff
@@ -0,0 +1,2168 @@
+_C.TASK.OBJECT_TO_GOAL_DISTANCE = CN()
+_C.TASK.OBJECT_TO_GOAL_DISTANCE.TYPE = "ObjectToGoalDistance"
+# -----------------------------------------------------------------------------
+# # OBJECT_DISTANCE_FROM_AGENT MEASUREMENT
+# -----------------------------------------------------------------------------
+_C.TASK.AGENT_TO_OBJECT_DISTANCE = CN()
+_C.TASK.AGENT_TO_OBJECT_DISTANCE.TYPE = "AgentToObjectDistance"
+
+from habitat.config.default import CN, Config
+
+# %%
+# @title Define `RearrangementTask` by extending `NavigationTask`
+from habitat.tasks.nav.nav import NavigationTask, merge_sim_episode_config
+
+
+def merge_sim_episode_with_object_config(
+    sim_config: Config, episode: Type[Episode]
+) -> Any:
+    sim_config = merge_sim_episode_config(sim_config, episode)
+    sim_config.defrost()
+    sim_config.objects = [episode.objects.__dict__]
+    sim_config.freeze()
+
+    return sim_config
+
+
+@registry.register_task(name="RearrangementTask-v0")
+class RearrangementTask(NavigationTask):
+    r"""Embodied Rearrangement Task
+    Goal: An agent must place objects at their corresponding goal position.
+    """
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+    def overwrite_sim_config(self, sim_config, episode):
+        return merge_sim_episode_with_object_config(sim_config, episode)
+
+
+# %% [markdown]
+# ## Implement a hard-coded and an RL agent
+#
+#
+
+# %%
+# @title Load the `RearrangementTask` in Habitat-Lab and run a hard-coded agent
+import habitat
+
+config_file = """
+ENVIRONMENT:
+  MAX_EPISODE_STEPS: 50
+  ITERATOR_OPTIONS:
+    SHUFFLE: True
+SIMULATOR:
+  TYPE: "RearrangementSim-v0"
+  ACTION_SPACE_CONFIG: "RearrangementActions-v0"
+  AGENT_0:
+    SENSORS: ['RGB_SENSOR']
+  RGB_SENSOR:
+    WIDTH: 256
+    HEIGHT: 256
+  DEPTH_SENSOR:
+    WIDTH: 256
+    HEIGHT: 256
+  GRAB_DISTANCE: 2.0
+  HABITAT_SIM_V0:
+    ENABLE_PHYSICS: True
+TASK:
+  TYPE: RearrangementTask-v0
+  SUCCESS_DISTANCE: 1.0
+  SENSORS: ['GRIPPED_OBJECT_SENSOR', 'OBJECT_POSITION', 'OBJECT_GOAL']
+  GOAL_SENSOR_UUID: object_goal
+  MEASUREMENTS: ['OBJECT_TO_GOAL_DISTANCE', 'AGENT_TO_OBJECT_DISTANCE']
+  POSSIBLE_ACTIONS: ['STOP', 'MOVE_FORWARD', 'GRAB_RELEASE']
+DATASET:
+  TYPE: RearrangementDataset-v0
+  SPLIT: train
+  DATA_PATH: data/datasets/rearrangement/coda/v1/{split}/{split}.json.gz
+
+"""
+with open("configs/tasks/rearrangement.yaml", "w") as f:
+    f.write(config_file)
+
+config = habitat.get_config("configs/tasks/rearrangement.yaml")
+
+
+def print_info(obs, metrics):
+    print(
+        "Gripped Object: {}, Distance To Object: {}, Distance To Goal: {}".format(
+            obs["gripped_object_id"],
+            metrics["agent_to_object_distance"],
+            metrics["object_to_goal_distance"],
+        )
+    )
+
+
+try:  # Got to make initialization idiot proof
+    env.close()
+except NameError:
+    pass
+
+env = habitat.Env(config)
```

Comment:
Since this is all one cell, maybe you should use with `habitat.Env(config) as env:` to demonstrate good practice on closing the environment when you are done with it. Should also make the colab more stable by closing the env on errors